cluster/src/main/java/io/atomix/cluster/impl/DefaultClusterMembershipService.java

```diff
@@ -259,11 +259,11 @@ private void activateMember(StatefulMember member) {
           localMember.rack(),
           localMember.host(),
           localMember.metadata())));
-    } else if (existingMember.getState() == State.INACTIVE) {
+    } else if (!existingMember.id().equals(localMember.id()) && existingMember.getState() == State.INACTIVE) {
```

Comment:
shouldn't we still post but not update the local map?

Reply:
This code is only hit when a remote node sends a message to the local node, in which case there could be a race (the remote node is sending old information about the local node). But what's actually needed is a separate check for the local node to ensure it fires an event.

Reply:
Never mind. This patch is all wrong. An event for the _local_ member's metadata is all that's needed.